salt/states/file.py

```diff
@@ -5564,7 +5570,16 @@ def keyvalue(
     try:
         with salt.utils.files.fopen(name, "r") as fd:
             file_contents = fd.readlines()
-    except OSError:
+            fd.close()
```

Comment:
Shouldn't the context manager close this file properly? Why are you adding the close call here?

Reply:
I just got caught up in the magnificent chaos of it all.